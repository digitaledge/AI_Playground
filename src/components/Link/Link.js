/* @flow */
import React from 'react';
import history from '../../history';

function isLeftClickEvent(event) {
  return event.button === 0;
}

function isModifiedEvent(event) {
  return !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey);
}

type Props = {
  to: string,
  children: any,
  onClick: ?Function,
};

class Link extends React.Component<Props> {
  static defaultProps = {
    onClick: null,
  };

  handleClick = event => {
    if (this.props.onClick) {
      this.props.onClick(event);
    }

    if (isModifiedEvent(event) || !isLeftClickEvent(event)) {
      return;
    }

    if (event.defaultPrevented === true) {
      return;
    }

    event.preventDefault();
    history.push(this.props.to);
  };

  render() {
    const { to, children, ...props } = this.props;
    return (
      <a href={to} {...props} onClick={this.handleClick}>
        {children}
      </a>
    );
  }
}

export default Link;
