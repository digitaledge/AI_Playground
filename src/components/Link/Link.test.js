/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */
import React from 'react';
import renderer from 'react-test-renderer';
import Link from './Link';

describe('Link', () => {
  test('Link does render width children', () => {
    const wrapper = renderer
      .create(
        <Link to="/" onClick={() => console.log('clicked')}>
          <div>Some link</div>
        </Link>,
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
