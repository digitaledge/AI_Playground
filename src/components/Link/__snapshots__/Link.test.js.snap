// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Link does render width children 1`] = `
<a
  href="/"
  onClick={[Function]}
>
  <div>
    Some link
  </div>
</a>
`;
